### PR TITLE
:sparkles: Speaker can see submission status: sbmitted, accepted, dec…

### PR DESCRIPTION
**Pull request by corinnekrych**
…lined

### CODE REVIEW

**Comment by corinnekrych** [comment]
This PR addresses: 

- [ ]  The talk status is updated in the speaker page: accepted, rejected

`confirmed` will be addressed in another PR to deal with speaker the confirmation step.

**Comment by corinnekrych** [comment]
unspecified-1

File: functions/triggers/onUpdateProposal.js [hunk 1]
```diff
@@ -18,33 +20,40 @@ module.exports = functions.firestore
     const rejected = talk.state === 'rejected'
     const { app, mailgun } = functions.config()
     if (!app) return Promise.reject(new Error('You must provide the app.url variable'))
-
+    const submissionUpdate = {
+      submissions: {},
+    }
+    submissionUpdate.submissions[eventId] = { ...talk }
```
Comment (bpetetot) [inline functions/triggers/onUpdateProposal.js]: You could directly write:
```js
const submissionUpdate = {
  submissions: {
    [eventId]: { ...talk },
  },
}
```

**Review by bpetetot — APPROVED** [review]
Seems great! thx 👍

File: functions/firestore/talk.js [hunk 2]
```diff
@@ -17,6 +17,15 @@ const updateTalk = (talkId, data) => admin
     { updateTimestamp: admin.firestore.FieldValue.serverTimestamp() }
   ))
 
+const partialUpdateTalk = (talkId, data) => admin
+  .firestore()
+  .collection('talks')
+  .doc(talkId)
+  .set(Object.assign(
+    data,
```
Comment (frinyvonnick) [inline functions/firestore/talk.js]: Here you mutate a parameter, I think it's not a good pratice. Maybe you could destructure it ?

File: functions/triggers/onUpdateProposal.js [hunk 3]
```diff
@@ -18,33 +20,40 @@ module.exports = functions.firestore
...
-        updateProposal(eventId, talk)]).then(([event, users]) => sendEmail(mailgun, {
+        updateProposal(eventId, talk)])
+        .then(([event, users]) => sendEmail(mailgun, {
           to: users.map(user => user.email),
           subject: `[${event.name}] Talk declined`,
-          html: talkRejected(event, users, talk, app.url),
-      }))
+          html: talkRejected(event, users, talk),
+        }))
+        .then(() => partialUpdateTalk(talk.id, submissionUpdate))
```
Comment (frinyvonnick) [inline functions/triggers/onUpdateProposal.js]: you don't use anything from updateProposal here maybe you could do it in parallel ?

File: src/store/reducers/data/talks.selector.js [hunk 4]
```diff
@@ -24,6 +24,29 @@ export const isSubmitted = (talkId, eventId) => (store) => {
   return false
 }
 
+const isEqualToStatus = status => (talkId, eventId) => (store) => {
+  const talk = store.data.talks.get(talkId)
+  if (talk && talk.submissions) {
```
Comment (frinyvonnick) [inline src/store/reducers/data/talks.selector.js]: Maybe we could use lodash here to simplify the code ? 
```js
if (has(talk, `submissions[${eventId}].state`)) {
  return talk.submissions[eventId].state === status
}
```

File: src/store/reducers/data/talks.selector.js [hunk 5]
```diff
@@ -24,6 +24,29 @@ export const isSubmitted = (talkId, eventId) => (store) => {
...
+  }
+  return false
+}
+
+/**
+ * Returns true if the talk has been accepted to the event
+ * @param {string} talkId talk id
+ * @param {string} eventId event id
+ */
+export const isAccepted = (talkId, eventId) => isEqualToStatus('accepted')(talkId, eventId)
```
Comment (frinyvonnick) [inline src/store/reducers/data/talks.selector.js]: I guess here 
```js
const isAccepted = (talkId, eventId) => isEqualToStatus('accepted')(talkId, eventId)
``` 
is the same as 
```js
const isAccepted = isEqualToStatus('accepted')
```

File: src/store/reducers/data/talks.selector.js [hunk 6]
```diff
@@ -24,6 +24,29 @@ export const isSubmitted = (talkId, eventId) => (store) => {
...
+
+/**
+ * Returns true if the talk has been declined to the event
+ * @param {string} talkId talk id
+ * @param {string} eventId event id
+ */
+export const isDeclined = (talkId, eventId) => isEqualToStatus('rejected')(talkId, eventId)
```
Comment (frinyvonnick) [inline src/store/reducers/data/talks.selector.js]: Same remark here 😄

**Comment by corinnekrych** [comment]
@bpetetot thanks for the review, all good comments I agreed with. I've fixed them.

**Review by frinyvonnick — APPROVED** [review]
Nice work 👍 ! We have very similar profile pictures but i'm not @bpetetot 😉

**Comment by corinnekrych** [comment]
@frinyvonnick Oh sorry i've missed that. Indeed get confused my the black background. Thanks Yvonnick!

**Comment by bpetetot** [comment]
Thanks @corinnekrych. Seems good to me 👍 I'm merging it in the feature branch

**Comment by bpetetot** [comment]
And thanks @frinyvonnick for the review

**Pull request merged**